### PR TITLE
Enhancement: CLI decoupling - Save analysis results in a directory

### PR DESCRIPTION
**Pull request by ayushnirwal**
## Description

This enhances the CLI tool's interface by adding a new argument `-d <path-to-dir>` or `--out-dir <path-to-dir>`. If this argument is provided the CLI tool will not open the analysis result in the dashboard. The results rather will be saved in the provided directory.

For single site input, the output directory will have the following files.
- `affected-cookies.csv`
- `cookies.csv`
- `report.csv`
- `technologies.csv`
- `out.json`

For multiple site input cases ( a sitemap or a CSV file ), the output directory will have directories for each URL with the files mentioned above.

## Testing Instructions

- Follow instructions on using CLI in the README.md.
- Create a directory for storing analysis result.
- Add a new argument for example - `npm run cli -- -u https://example.com -np -d ./foo`.
- The directory `foo` should have the files mentioned in the description of the PR.

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Partially Addresses #203

### CODE REVIEW

**Comment by mohdsayed** [comment]
For later, one idea would be to also include the `.html` file report, similar to the lighthouse CLI.

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -18,22 +18,26 @@
...
 import Utility from './utils/utility';
-import { fetchDictionary } from './utils/fetchCookieDictionary';
 import { analyzeCookiesUrlsInBatches } from './procedures/analyzeCookieUrlsInBatches';
 import { analyzeTechnologiesUrlsInBatches } from './procedures/analyzeTechnologiesUrlsInBatches';
-import { delay } from './utils';
+import {
+  fetchDictionary,
+  delay,
+  getUrlListFromArgs,
+  validateArgs,
```
Comment (mohdsayed) [inline packages/cli/src/index.ts:38]: 👍

File: packages/cli-dashboard/tsconfig.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
   "include": [
     "src",
     "../../node_modules/@types/jest/index.d.ts",
-    "src/**/*.json"
+    "src/**/*.json",
+    "../common/src/utils/generateReports"
```
Comment (mayan-000) [inline packages/cli-dashboard/tsconfig.json:10]: Seems an auto include, we can remove it.

File: packages/cli/src/index.ts [hunk 3]
```diff
@@ -380,22 +214,19 @@ const getUrlListFromArgs = async (
...
-  await writeFile(
-    outputDir + '/out.json',
-    JSON.stringify(url ? result[0] : result, null, 4)
-  );
-
-  exec('npm run cli-dashboard:dev');
+  await saveResults(path.join(outputDir, prefix), result);
 
-  await delay(2000);
+  if (outDir) {
```
Comment (mayan-000) [inline packages/cli/src/index.ts:222]: Can we early return?

File: packages/cli/src/utils/getUrlListFromArgs.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  url: string,
+  sitemapUrl: string,
+  csvPath: string,
+  sitemapPath: string,
+  // @ts-ignore Package does not support typescript.
+  spinnies
+): Promise<string[]> => {
+  let urls: string[] = [];
+
+  if (url) {
```
Comment (mayan-000) [inline packages/cli/src/utils/getUrlListFromArgs.ts:46]: Can we break the processing of each arg into smaller functions inside the same file?

File: packages/cli/src/utils/validateArgs.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+ */
+// eslint-disable-next-line complexity
+const validateArgs = async (
+  url: string,
+  sitemapUrl: string,
+  csvPath: string,
+  sitemapPath: string,
+  numberOfUrls: string,
+  outDir: string
+) => {
```
Comment (mayan-000) [inline packages/cli/src/utils/validateArgs.ts:38]: How about 
const numArgs = [Boolean(url), Boolean(sitemapURL), Boolean(csvPath), Boolean(sitemapPath)].reduce((acc, arg) => {
    acc += arg ? 1 : 0;
}, 0);

File: packages/cli/src/utils/validateArgs.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  ) {
+    console.log(
+      `Please provide ONLY one of the following 
+        a) URL of a site (-u or --url) 
+        b) URL of a sitemap (-s or --sitemap-url)
+        c) Path to a CSV file (-c or --csv-path)
+        d) Path to a XML file (-p or --sitemap-path)`
+    );
+    process.exit(1);
+  }
```
Comment (mayan-000) [inline packages/cli/src/utils/validateArgs.ts:64]:
```suggestion
const numArgs = [Boolean(url), Boolean(sitemapURL), Boolean(csvPath), Boolean(sitemapPath)].reduce((acc, arg) => {
acc += arg ? 1 : 0;
}, 0);

  if (numArgs === 0) {
    console.log(
      `Please provide one of the following 
        a) URL of a site (-u or --url) 
        b) URL of a sitemap (-s or --sitemap-url)
        c) Path to a CSV file (-c or --csv-path)
        d) Path to a XML file (-p or --sitemap-path)`
    );
    process.exit(1);
  }

  if (numArgs > 1) {
    console.log(
      `Please provide ONLY one of the following 
        a) URL of a site (-u or --url) 
        b) URL of a sitemap (-s or --sitemap-url)
        c) Path to a CSV file (-c or --csv-path)
        d) Path to a XML file (-p or --sitemap-path)`
    );
    process.exit(1);
  }
```

**Review by mohdsayed — APPROVED** [review]

**Pull request merged**